packages/typespec-client-generator-core/test/http/method-parameter-segments.test.ts

```diff
@@ -0,0 +1,187 @@
+  strictEqual(queryParam.methodParameterSegments[0][0].name, "input");
+  strictEqual(queryParam.methodParameterSegments[0][0].kind, "method");
+  strictEqual(queryParam.methodParameterSegments[0][1].name, "q");
+  strictEqual(queryParam.methodParameterSegments[0][1].kind, "property");
+});
+
+it("spread body - multiple paths", async () => {
+  await runner.compileWithBuiltInService(`
+    model Input {
+      key1: string;
+      key2: string;
+    }
+    op myOp(...Input): void;
+    `);
+  const sdkPackage = runner.context.sdkPackage;
+  const method = getServiceMethodOfClient(sdkPackage);
+  const serviceOperation = method.operation;
+
+  ok(serviceOperation.bodyParam);
+
+  // Check correspondingMethodParams returns multiple final properties
+  strictEqual(serviceOperation.bodyParam.correspondingMethodParams.length, 2);
+
+  // Check new methodParameterSegments has multiple paths (one per spread property)
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments.length, 2);
+
+  // First path should be to key1
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments[0].length, 1);
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments[0][0].name, "key1");
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments[0][0].kind, "method");
+
+  // Second path should be to key2
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments[1].length, 1);
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments[1][0].name, "key2");
+  strictEqual(serviceOperation.bodyParam.methodParameterSegments[1][0].kind, "method");
+});
+
+it("deeply nested property path", async () => {
+  await runner.compileWithBuiltInService(`
+    model Level2 {
+      @query q: string;
+    }
+    model Level1 {
+      level2: Level2;
+    }
+    op myOp(input: Level1): void;
+    `);
+  const sdkPackage = runner.context.sdkPackage;
+  const method = getServiceMethodOfClient(sdkPackage);
+  const serviceOperation = method.operation;
+
+  strictEqual(serviceOperation.parameters.length, 2);
+  const queryParam = serviceOperation.parameters[0];
+
+  // Check correspondingMethodParams returns only the final property
+  strictEqual(queryParam.correspondingMethodParams.length, 1);
+  strictEqual(queryParam.correspondingMethodParams[0].name, "q");
+
+  // Check new methodParameterSegments shows complete path through all levels
+  strictEqual(queryParam.methodParameterSegments.length, 1);
+  strictEqual(queryParam.methodParameterSegments[0].length, 3);
+  strictEqual(queryParam.methodParameterSegments[0][0].name, "input");
+  strictEqual(queryParam.methodParameterSegments[0][0].kind, "method");
+  strictEqual(queryParam.methodParameterSegments[0][1].name, "level2");
+  strictEqual(queryParam.methodParameterSegments[0][1].kind, "property");
+  strictEqual(queryParam.methodParameterSegments[0][2].name, "q");
+  strictEqual(queryParam.methodParameterSegments[0][2].kind, "property");
```

Comment:
Is this correct?

I'd expect client method be `myOp(Level1: level1)` without any `q` in method param.

Reply:
The client level method is `op myOp(input: Level1): void`. The HTTP level `q` parameter's method parameter mapping is `input`-> `level2` -> `q`.